### PR TITLE
Add support for Scala 2.13.3

### PR DESCRIPTION
**Pull request by tgodzik**

### CODE REVIEW

**Comment by tgodzik** [comment]
I see 3 issues currently:
- scalameta regression - already fixed
- Bloop fails to compile 2.13.3 - this is clocking unfortunately
- ammonite is not yet released for 2.13.3, but we can just ignore the test for now

**Comment by kpbochenek** [comment]
Bloop issue: https://github.com/scalacenter/bloop/issues/1330

**Comment by kpbochenek** [comment]
bloop fix is done and will be delivered in `1.4.3` release(hopefully soon :crossed_fingers: ).
ammonite with support for 2.13.3: `  val ammonite = "2.1.4-8-5d0c097"` (Ammonite213Suite passes)

File: build.sbt [hunk 1]
```diff
@@ -186,12 +186,13 @@ lazy val V = new {
...
   def deprecatedScala2Versions = Seq(scala211, "2.12.8", "2.12.9", "2.13.0")
-  def nonDeprecatedScala2Versions = Seq(scala213, scala212, "2.12.10", "2.13.1")
+  def nonDeprecatedScala2Versions =
+    Seq(scala213, scala212, "2.12.10", "2.13.1", "2.13.2")
   def scala2Versions = nonDeprecatedScala2Versions ++ deprecatedScala2Versions
 
   // Scala 3
   def nonDeprecatedScala3Versions = Seq(scala3, "0.24.0")
-  def deprecatedScala3Versions = Seq()
+  def deprecatedScala3Versions = Seq("0.24.0-RC1")
```
Comment (tgodzik) [inline build.sbt:195]: Added this back, since I figured it we should not be just dropping it on users.

Comment (tgodzik) [reply build.sbt]: Also added 0.23.0, which doesn't influence the tests

File: mtags/src/main/scala-2/scala/meta/internal/pc/completions/FilenameCompletions.scala [hunk 2]
```diff
@@ -7,11 +7,8 @@ import scala.collection.immutable.Nil
 import scala.util.control.NonFatal
 
 import scala.meta.internal.pc.CompletionFuzzy
-import scala.meta.internal.pc.CompletionFuzzy
```
Comment (ckipp01) [inline mtags/src/main/scala-2/scala/meta/internal/pc/completions/FilenameCompletions.scala:10]: Random side note, the next release of OrganizeImports will also remove duplicated ones like this 🥳

File: build.sbt [hunk 3]
```diff
@@ -186,12 +186,13 @@ lazy val V = new {
...
   def deprecatedScala2Versions = Seq(scala211, "2.12.8", "2.12.9", "2.13.0")
-  def nonDeprecatedScala2Versions = Seq(scala213, scala212, "2.12.10", "2.13.1")
+  def nonDeprecatedScala2Versions =
+    Seq(scala213, scala212, "2.12.10", "2.13.1", "2.13.2")
   def scala2Versions = nonDeprecatedScala2Versions ++ deprecatedScala2Versions
 
   // Scala 3
   def nonDeprecatedScala3Versions = Seq(scala3, "0.24.0")
-  def deprecatedScala3Versions = Seq()
+  def deprecatedScala3Versions = Seq("0.23.0", "0.24.0-RC1")
```
Comment (ckipp01) [inline build.sbt:195]: Wait, actually, with adding 0.23.0 back in here, don't we also need the `scala-0.23/` directory with ClassFinder in it? If not you get the following when trying to cross publish:
```
[info] Setting Scala version to 0.23.0 on 2 projects.
[info] Excluded 10 projects, run ++ 0.23.0 -v for more details.
[info] Reapplying settings...
[info] set current project to default-86938c (in build file:/Users/ckipp/Documents/scala-workspace/scalameta-org/metals/)
[info] Wrote /Users/ckipp/Documents/scala-workspace/scalameta-org/metals/mtags/target/scala-0.23.0/mtags_0.23.0-0.9.1-SNAPSHOT.pom
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] Compiling 51 Scala sources to /Users/ckipp/Documents/scala-workspace/scalameta-org/metals/mtags/target/scala-0.23.0/classes ...
[error] -- [E008] Member Not Found Error: /Users/ckipp/Documents/scala-workspace/scalameta-org/metals/mtags/src/main/scala-3/scala/meta/internal/pc/CompilerInterfaces.scala:15:34
[error] 15 |import dotty.tools.dotc.reporting.Diagnostic.Error
[error]    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]    |value Diagnostic is not a member of dotty.tools.dotc.reporting - did you mean reporting.diagnostic?
[error] -- [E006] Unbound Identifier Error: /Users/ckipp/Documents/scala-workspace/scalameta-org/metals/mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala:523:8
[error] 523 |        ClassFinder.findClassForOffset(params.offset, filename)(
[error]     |        ^^^^^^^^^^^
[error]     |        Not found: ClassFinder
[error] two errors found
[error] (mtags / Compile / compileIncremental) Compilation failed
```

Comment (tgodzik) [reply build.sbt:195]: och yeah, forgot about :(

Comment (tgodzik) [reply build.sbt]: I will remove 0.23 back and add only 0.24.0-RC1 to deprecated. With the next upgrade I will try to keep both versions.

File: mtags/src/main/scala-2/scala/meta/internal/pc/completions/FilenameCompletions.scala (hunk 2 above)
Comment (tgodzik) [reply mtags/src/main/scala-2/scala/meta/internal/pc/completions/FilenameCompletions.scala:10]: I think that was actually removed by that rule :sweat_smile:

Comment (tgodzik) [reply mtags/src/main/scala-2/scala/meta/internal/pc/completions/FilenameCompletions.scala:10]: Or maybe it was due to warnings I saw?

Comment (ckipp01) [reply mtags/src/main/scala-2/scala/meta/internal/pc/completions/FilenameCompletions.scala:10]: O really? Haha, nevermind then. I thought you did it manually.

**Comment by alexandru** [comment]
W00t! When are you going to make a release?

**Comment by ckipp01** [comment]
> W00t! When are you going to make a release?

One was made last night 😉.
https://scalameta.org/metals/blog/2020/07/01/lithium.html

**Comment by gabro** [comment]
@alexandru 9h ago 😂 https://twitter.com/scalameta/status/1278450415424671745?s=21

@tgodzik is too fast to follow

**Comment by alexandru** [comment]
Awesome, thanks ❤️

**Pull request merged**